pkg/transform/oauth/github.go

```diff
@@ -66,3 +67,32 @@ func buildGitHubIP(serializer *json.Serializer, p IdentityProvider) (IdentityPro
 
 	return idP, *secret, caConfigmap, nil
 }
+
+// ValidateGithubProvider validate github identity provider
+func ValidateGithubProvider(serializer *json.Serializer, p IdentityProvider) error {
```

Comment:
Having specific validation methods for each specific provider seems reasonable. It will definitely be easy to maintain.